src/main/java/au/com/dmg/fusion/util/PairingData.java

```diff
@@ -0,0 +1,127 @@
+package au.com.dmg.fusion.util;
+
+import com.squareup.moshi.Json;
+
+import java.util.Random;
+
+public class PairingData {
+    @Json(name = "s")
+    private final String saleID;
+    @Json(name = "p")
+    private final String pairingPOIID;
+    @Json(name = "k")
+    private final String kek;
+    @Json(name = "c")
+    private final String certificationCode;
+    @Json(name = "n")
+    private final String posName;
+    @Json(name = "v")
+    private final Integer version;
+
+    public String getSaleID(){return saleID; }
+    public String getPairingPOIID(){ return pairingPOIID; }
+    public String getKek() {return kek;}
+    public String getCertificationCode(){return certificationCode;}
+    public String getPosName(){return posName; }
+    public Integer getVersion(){return version; }
+
+    public static class Builder {
+        private String saleID;
+        private String pairingPOIID;
+        private String kek;
+        private String certificationCode;
+        private String posName;
+        private Integer version;
+
+        public Builder() {
+        }
+
+        Builder(String saleID, String pairingPOIID, String kek, String certificationCode, String posName, Integer version){
+            this.saleID = saleID;
+            this.pairingPOIID = pairingPOIID;
+            this.kek = kek;
+            this.certificationCode = certificationCode;
+            this.posName = posName;
+            this.version = version;
+        }
+
+        private Builder saleID(String saleID){
+            this.saleID = saleID;
+            return Builder.this;
+        }
+        private Builder pairingPOIID(String pairingPOIID){
+            this.pairingPOIID = pairingPOIID;
+            return Builder.this;
+        }
+        private Builder kek(String kek){
+            this.kek = kek;
+            return Builder.this;
+        }
+        private Builder certificationCode(String certificationCode){
+            this.certificationCode = certificationCode;
+            return Builder.this;
+        }
+        private Builder posName(String posName){
+            this.posName = posName;
+            return Builder.this;
+        }
+        private Builder version(Integer version){
+            this.version = version;
+            return Builder.this;
+        }
+
+        public PairingData build(){
+            if (this.saleID == null) {
```

Comment:
Please add checking for empty string too.  Please add the same checking (not empty) to the other fields.  Since they are required, they should be both _not null_ and _not empty_.